scripts/core/device.yml

```diff
@@ -785,3 +785,15 @@ etors:
     - name: ATOMIC_CONCURRENT_ACCESS
       desc: "Memory can be accessed atomically and concurrently"
       value: "$X_BIT(3)"
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Return true if the device supports reading from and writing to host pipes."
+class: $xDevice
+name: HostPipeRWSupported
+ordinal: "0"
+details:
+    - "x"
+params:
+    - type: $x_device_handle_t
+      name: hDevice
+      desc: "[in] handle of the device instance"
```

Comment:
The way the generator is set up I don't think its possible to specify anything other than `ur_result_t`, so you'd need to have an our parameter to tell the user the answer to this query.

Rather than a new entry point, I think this should be queriable via [urDeviceGetInfo](https://oneapi-src.github.io/unified-runtime/core/api.html#urdevicegetinfo) by adding a new enumeration to [ur_device_info_t](https://oneapi-src.github.io/unified-runtime/core/api.html#_CPPv416ur_device_info_t), e.g. `UR_DEVICE_INFO_HOST_PIPE_RW_SUPPORTED` with an out parameter type of `ur_bool_t`.

Reply:
Ah I was just assuming I could make it return true/false. That's a great suggestion thanks I'll update it to work like that!